core/domain/exp_jobs_one_off.py

```diff
@@ -542,3 +543,75 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class ExplorationMigrationValidationJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that migrates the html content of an exploration into the valid
+    Textangular format and validates it.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_services.get_exploration_from_model(item)
+        # err_dict is a dictionary to store the invalid tags and the
+        # invalid parent-child relations that we find.
+        err_dict = {}
+
+        # All the invalid html strings will be stored in this.
+        err_dict['strings'] = []
+
+        allowed_parent_list = (
+            feconf.RTE_CONTENT_SPEC['RTE_TYPE_TEXTANGULAR']
+            ['ALLOWED_PARENT_LIST'])
+        allowed_tag_list = (
+            feconf.RTE_CONTENT_SPEC['RTE_TYPE_TEXTANGULAR']
+            ['ALLOWED_TAG_LIST'])
+
+        html_list = exploration.get_all_html_content_strings()
+
+        for html_data in html_list:
+            migrated_data = html_cleaner.convert_to_text_angular(html_data)
+            soup = bs4.BeautifulSoup(migrated_data, 'html.parser')
```

Comment:
It might not be a bad idea to extract all this stuff into an html_cleaner.validate_textangular_format() function, so you can use it in multiple places without having to copy it all over the place.

Reply:
Done!